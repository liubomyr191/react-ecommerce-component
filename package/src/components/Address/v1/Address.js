import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { addTypographyStyles, CustomPropTypes } from "../../../utils";

const AddressElement = styled.address`
  ${addTypographyStyles("Address", "bodyText")};
`;

class Address extends Component {
  static propTypes = {
    /**
     * Address data
     */
    address: CustomPropTypes.address.isRequired,
    /**
     * You can provide a `className` prop that will be applied to the outermost DOM element
     * rendered by this component. We do not recommend using this for styling purposes, but
     * it can be useful as a selector in some situations.
     */
    className: PropTypes.string,
    /**
     * If you've set up a components context using
     * [@reactioncommerce/components-context](https://github.com/reactioncommerce/components-context)
     * (recommended), then this prop will come from there automatically. If you have not
     * set up a components context or you want to override one of the components in a
     * single spot, you can pass in the components prop directly.
     */
    components: PropTypes.shape({}).isRequired
  };

  static defaultProps = {};

  render() {
    const { address, className } = this.props;

    return (
      <AddressElement className={className}>
        {address.fullName}
        <br />
        {address.address1}
        <br />
        {address.address2 !== null && address.address2 !== "" ? (
          <span>
            {address.address2} <br />
          </span>
        ) : null}
        {address.city}, {address.region} {address.postal} <br />
        {address.country}
      </AddressElement>
    );
  }
}

export default Address;
