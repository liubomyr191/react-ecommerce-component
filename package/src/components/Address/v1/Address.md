### Overview

### Usage

Document various live component examples here. See https://react-styleguidist.js.org/docs/documenting.html

```jsx
const address = {
  _id: "1",
  address1: "7742 Hwy 23",
  address2: "",
  country: "US",
  city: "Belle Chasse",
  fullName: "Salvos Seafood",
  postal: "70037",
  region: "LA",
  phone: "(504) 393-7303"
};

<Address address={address} />

```

**With invalid address properties**

```jsx
const address = {
  _id: "1",
  address1: "7742 Hwy 23",
  address2: "",
  country: "US",
  city: "Belle Chasse",
  fullName: "Salvos Seafood",
  postal: "70037",
  region: "LA",
  phone: "(504) 393-7303"
};

const invalidAddressProperties = ["fullName", "address1", "region"];

<Address address={address} invalidAddressProperties={invalidAddressProperties} />

```

### Theme

See [Theming Components](./#!/Theming%20Components).

#### Typography

- The address preview text uses `bodyText` style with `rui_components.Address` override
