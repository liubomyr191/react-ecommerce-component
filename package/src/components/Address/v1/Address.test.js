import React from "react";
import renderer from "react-test-renderer";
// import { shallow } from "enzyme";
import Address from "./Address";

const mockAddress = {
  _id: "1",
  address1: "7742 Hwy 23",
  address2: "",
  country: "US",
  city: "Belle Chasse",
  fullName: "Salvos Seafood",
  postal: "70037",
  region: "LA",
  phone: "(504) 393-7303"
};

test("basic snapshot", () => {
  const component = renderer.create(<Address address={mockAddress} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
