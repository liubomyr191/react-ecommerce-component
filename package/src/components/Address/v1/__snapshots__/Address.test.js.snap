// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
}

<address
  className="c0"
>
  Salvos Seafood
  <br />
  7742 Hwy 23
  <br />
  Belle Chasse
  , 
  LA
   
  70037
   
  <br />
  US
</address>
`;
